File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> st = new Stack<>();
        int n = s.length();
        st.push(s.charAt(0));
        for(int i = 1; i < n; i++){
            char ch = s.charAt(i);
            if(st.size() > 0 && st.peek() != ch) st.push(ch);
            else if(st.size() == 0) st.push(ch);
            else st.pop();
        }
        StringBuilder ans = new StringBuilder();
        while(st.size() > 0) ans.append(st.pop()) ;
        ans.reverse();
        return ans.toString();
    }
}

File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {
    let arr = [];
    for(let i = 0; i < s.length; i++){
        let ch = s.charAt(i);
        if(arr.length > 0 &&  ch === arr[arr.length-1]){
            arr.pop();
        }
        else{
            arr.push(ch);
        }
    }
    return arr.join('');
};